src/endpoint.rs

```diff
@@ -41,7 +41,10 @@ const PORT_FORWARD_TIMEOUT: u64 = 30;
 const ECHO_SERVICE_QUERY_TIMEOUT: u64 = 30;
 
 /// Standard size of our channel bounds
-const STANDARD_CHANNEL_SIZE: usize = 1000;
+const STANDARD_CHANNEL_SIZE: usize = 10000;
+
+/// Max number of attempts for connection retries
+const MAX_ATTEMPTS: usize = 10;
```

Comment:
Would `10` be too much? Meaning, it can take some time to complete all 10 retries if there's the receiver is gone surely. Maybe a smaller number like 3 or 5?